src/mechanics/dofs/DofMatrixContainer.h

```diff
@@ -80,6 +81,15 @@ class DofMatrixContainer
         return out;
     }
 
+    std::vector<DofType> DofTypes() const
+    {
+        std::vector<DofType> dofTypes;
+        for (const auto& data : mData)
+            if (data.first.first.Id() == data.first.second.Id())
+                dofTypes.push_back(data.first.first);
```

Comment:
Returns only dofs that occur in pairs? Is this right?

Reply:
Well, maybe I should a comment on that on the code. 
~~~
//! @remark std::vector allows duplicates and we do not want duplicates. So we 
//! extract the dofs types from the diagonal entries, where they only appear once.
~~~
Does that make sense?